### ISSUE TITLE
[BUG] Charts pages are loading very slow

### ISSUE THREAD

**Issue by dscham** [rancher/rancher#44323]
**Rancher Server Setup**
- Rancher version: 2.8.1
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE2
- Proxy/Cert Details: No Proxy, access via WireGuard, default Certs

**Information about the Cluster**
- Kubernetes version: v1.26.12+rke2r1
- Cluster Type (Local/Downstream): local
- Custom: Running 1 controller and 2 worker nodes on a single Proxmox host
  - Controller Node: 8 GiB RAM, 4 Cores
  - Worker Nodes: 4 GiB RAM, 2 Cores

**User Information**
- What is the role of the user logged in? Default Admin

**Describe the bug**
Some pages, especially those loading helm charts (Installed Apps, Charts or Chart Details page) are loading very slow.

**To Reproduce**
- Fresh Ubuntu22.04 Cloudimage
- Installed iptables and nfs-common on nodes
- Installed RKE2, according to docs and with no customization
- Installed Rancher via helm, according to docs and with a resolvable hostname=

**Result**
- All Nodes are Ready
- All pods are in Running state
- Some pages, especially those in the Apps menu, are loading very slow (couple KiB/s)

**Expected Result**
- Fast page loads (couple seconds, not a minute or even minutes)

**Additional context**
I set up a WireGuard connection to the Proxmox host via pfSense (also running on Proxmox). There are no proxies in between. There are no firewall rules, to my knowledge, between my workstation and the host. Nodes are on a VLAN (42) and have subnet 10.0.42.0/24. All other services on the Proxmox host (pfSense and a Ghost container, ssh to the nodes and ohter vms) are snappy as they should be.

I've been scouring the internet for 2 hours to find similar issues. Found some, but all that i found where due to failing pods or wrong DNS config. Which I don't have.

It is not a client side problem (i cant see any issues in Performance on dev tools). It's not a network problem, iperf is good (150 MBit up, 50 down). But for some pages, loads are on the order of KiB/s. (Judging by eye on the Networks dev tools tab)

I have helm installed on the controller node itself, which is also performing normally. Actually, everything is running normally. Except the Rancher UI. Which is sometimes loading quick, and sometimes very slow. I haven't found any pattern yet.

Edit: added more info about setup

**Comment by torchiaf** [comment]
Thanks @dscham, it seems to be related to a UI performance issue. We are still working on a performance improvement.

**Comment by dscham** [comment]
Thanks for the reply @torchiaf! 
Ah yes, I stumbled upon an issue having a checklist for that, but can't find it anymore. You don't happen to have a link, do you?

**Comment by torchiaf** [comment]
@richard-cox maybe you have the list of performance-related issues.

**Comment by richard-cox** [comment]
This might not be UI performance related, possibly core Rancher.

The pages referenced will fetch charts from help repos added to the cluster. These look something like `/k8s/clusters/c-m-n4x45x4b/v1/catalog.cattle.io.clusterrepos/rancher-partner-charts?link=index`. Are those requests taking a long time to complete and is the time spent more waiting for server or download?

![image](https://github.com/rancher/rancher/assets/18697775/1ef885b0-a654-4826-b4d6-083a3107d609)

**Comment by dscham** [comment]
Thanks @richard-cox!

I exported a HAR file from going to the login page (cluster dashboard) to fully loaded Apps/Charts:
[rancher.dslan.at.zip](https://github.com/rancher/rancher/files/14192196/rancher.dslan.at.zip)

Probably more effective than screenshots.

I had htop running on the nodes during the request. One of the workers peaked single core a bit a couple of times, but never for more than a second.

**Comment by richard-cox** [comment]
@dscham Thanks, grabbed the file. Probably not best in the public sphere though

**Comment by dscham** [comment]
I guess that's right. Although I have nothing really private in those afaik. And the install isn't reachable from the internet for now anyway.

**Comment by richard-cox** [comment]
The initial response times are generally very slow, with time most spent waiting for server.

There's some long running requests to fetch schemas. There is a performance improvement coming in 2.9.0 that reduces the payload. 

For charts, there's three that took over a minute downloading content. 

Bit of everything there, but unfortunately they don't look UI related where Francesco and I reside

**Comment by dscham** [comment]
Thanks!

That's what I gathered too and why I posted an issue about it. Otherwise I'd just left it be, and waited for updates. This seems to go deeper though. I already tested network problems to the best of my ability. Everything looks fine. And since htop doesn't look like everything is peaking all the time, I don't believe the nodes are underpowered either.

So something weird is going on there, right?

**Comment by richard-cox** [comment]
I think so. Looking at the size of responses none seem large enough to warrant the long requests.

Next step would be to check out the upstream cattle-system/rancher pods to check for errors. I wouldn't be able to assist there though

**Comment by dscham** [comment]
I'll have a look at the container logs.

**Comment by dscham** [comment]
I ran kubectl logs on the rancher pod and couldn't see anything suspicious either.

Interestingly though, while loading the Charts page, the output of the logs command took quite a bit longer too. It seems all pages take longer or are stuck during that.

I had htop running in another terminal the whole time, which didn't show more than 20% CPU usage and no more than half the RAM used. I then downloaded hdparm, because I thought it may be disk related. But I get reads and writes in the GiB/s, expected of NVME.

**Comment by dscham** [comment]
I really only notice big issues with the Apps pages. So big in fact, all tabs with my Rancher URL in them are getting stuck while the Charts are loading. Even Longhorn, opened via Rancher UI.

I just installed Authentik to the cluster, and page performance is great there. Even while loading Charts. 

What I realise now is. When using `helm`, installed on the node manually before Rancher, I can't seem to find the same repos or installed releases I get when in the Rancher UI. Don't know if this is connected somehow. I exported the rke2.yaml to `KUBECONFIG`. And also tried adding the `--kubeconfig` argument for that. Not even getting the same repos listed with `helm repo list`
